File: src/ice/graphics/SglDrawConst.java
package ice.graphics;

import arc.Core;
import arc.graphics.g2d.TextureRegion;
import arc.scene.style.Drawable;
import arc.scene.style.TextureRegionDrawable;
import arc.util.Tmp;
import mindustry.gen.Tex;
import mindustry.graphics.Pal;

public class SglDrawConst {
  public static final TextureRegion[] EMP_REGIONS = new TextureRegion[0];


  //Texture regions
  public static Drawable transparent, sglLaunchLogo, squareMarker, matrixArrow, sglIcon, artistIcon, codeIcon, translateIcon,
      soundsIcon, time, techPoint, inspire,

      startIcon, databaseIcon, publicInfoIcon, aboutIcon, configureIcon, contributeIcon, debuggingIcon, nuclearIcon, matrixIcon,

      qqIcon, telegramIcon,
      showInfos, unShowInfos, showRange, hold, defaultShow,
      grayUI, padGrayUI, darkgrayUI,
      grayUIAlpha, padGrayUIAlpha, darkgrayUIAlpha,
      sgl2, a_z;

  public static TextureRegion cursor;
  public static <T extends Drawable> T getModDrawable(String name) {
    return Core.atlas.getDrawable( "ice-" + name);
  }
  public static void load(){
   transparent = getModDrawable("transparent");
    sglLaunchLogo = getModDrawable("launch_logo");
    squareMarker = getModDrawable("square_marker");
    matrixArrow = getModDrawable("matrix_arrow");
    sglIcon = getModDrawable("sgl_icon");
    artistIcon = getModDrawable("artist");
    codeIcon = getModDrawable("code");
    translateIcon = getModDrawable("translate");
    soundsIcon = getModDrawable("sound");
    time = getModDrawable("time");
    techPoint = getModDrawable("tech_point");
    inspire = getModDrawable("inspire");
    startIcon = getModDrawable("icon_start");
    databaseIcon = getModDrawable("icon_database");
    publicInfoIcon = getModDrawable("icon_publicInfo");
    aboutIcon = getModDrawable("icon_about");
    configureIcon = getModDrawable("icon_configure");
    contributeIcon = getModDrawable("icon_contribute");
    debuggingIcon = getModDrawable("debugging");
    nuclearIcon = getModDrawable("nuclear");
    matrixIcon = getModDrawable("matrix");
    qqIcon = getModDrawable("qq");
    telegramIcon = getModDrawable("telegram");
    showInfos = getModDrawable("show_infos");
    unShowInfos = getModDrawable("unshow_infos");
    showRange = getModDrawable("show_range");
    hold = getModDrawable("hold");
    defaultShow = getModDrawable("default_show");
    sgl2 = getModDrawable("sgl-2");
    a_z = getModDrawable("a_z");

 //   cursor = getModAtlas("cursor");

    grayUI = ((TextureRegionDrawable) Tex.whiteui).tint(Pal.darkerGray);
    padGrayUI = ((TextureRegionDrawable) Tex.whiteui).tint(Pal.darkerGray);
    padGrayUI.setLeftWidth(8);
    padGrayUI.setRightWidth(8);
    padGrayUI.setTopHeight(8);
    padGrayUI.setBottomHeight(8);
    darkgrayUI = ((TextureRegionDrawable) Tex.whiteui).tint(Pal.darkestGray);

    grayUIAlpha = ((TextureRegionDrawable) Tex.whiteui).tint(Tmp.c1.set(Pal.darkerGray).a(0.7f));
    padGrayUIAlpha = ((TextureRegionDrawable) Tex.whiteui).tint(Tmp.c1.set(Pal.darkerGray).a(0.7f));
    padGrayUIAlpha.setLeftWidth(8);
    padGrayUIAlpha.setRightWidth(8);
    padGrayUIAlpha.setTopHeight(8);
    padGrayUIAlpha.setBottomHeight(8);
    darkgrayUIAlpha = ((TextureRegionDrawable) Tex.whiteui).tint(Tmp.c1.set(Pal.darkestGray).a(0.7f));
  }
}

File: src/singularity/graphic/SglDrawConst.kt
package singularity.graphic

import arc.graphics.Color
import arc.graphics.g2d.TextureRegion
import arc.scene.style.Drawable
import arc.scene.style.TextureRegionDrawable
import arc.util.Tmp
import mindustry.gen.Tex
import mindustry.graphics.Pal
import singularity.Singularity

object SglDrawConst {
  @JvmField val EMP_REGIONS: Array<TextureRegion?> = arrayOfNulls<TextureRegion>(0)

  //Colors
  @JvmStatic val transColor: Color = Color(0f, 0f, 0f, 0f)
  @JvmStatic val fexCrystal: Color = Color.valueOf("FF9584")
  @JvmStatic val matrixNet: Color = Color.valueOf("D3FDFF")
  @JvmStatic val matrixNetDark: Color = Color.valueOf("9ECBCD")
  @JvmStatic val ion: Color = Color.valueOf("#D1D19F")
  @JvmStatic val dew: Color = Color.valueOf("ff6214")
  @JvmStatic val frost: Color = Color.valueOf("AFF7FF")
  @JvmStatic val winter: Color = Color.valueOf("6CA5FF")

  //Text colors
  const val COLOR_ACCENT: String = "[accent]"

  //Texture regions
  lateinit var transparent: Drawable
  lateinit var sglLaunchLogo: Drawable
  lateinit var squareMarker: Drawable
  lateinit var matrixArrow: Drawable
  lateinit var sglIcon: Drawable
  lateinit var artistIcon: Drawable
  lateinit var codeIcon: Drawable
  lateinit var translateIcon: Drawable
  lateinit var soundsIcon: Drawable
  lateinit var time: Drawable
  lateinit var techPoint: Drawable
  lateinit var inspire: Drawable

  lateinit var startIcon: Drawable
  lateinit var databaseIcon: Drawable
  lateinit var publicInfoIcon: Drawable
  lateinit var aboutIcon: Drawable
  lateinit var configureIcon: Drawable
  lateinit var contributeIcon: Drawable
  lateinit var debuggingIcon: Drawable
  lateinit var nuclearIcon: Drawable
  lateinit var matrixIcon: Drawable

  lateinit var qqIcon: Drawable
  lateinit var telegramIcon: Drawable
  lateinit var showInfos: Drawable
  lateinit var unShowInfos: Drawable
  lateinit var showRange: Drawable
  lateinit var hold: Drawable
  lateinit var defaultShow: Drawable
  lateinit var grayUI: Drawable
  lateinit var padGrayUI: Drawable
  lateinit var darkgrayUI: Drawable
  lateinit var grayUIAlpha: Drawable
  lateinit var padGrayUIAlpha: Drawable
  lateinit var darkgrayUIAlpha: Drawable
  lateinit var sgl2: Drawable
  lateinit var a_z: Drawable

  var cursor: TextureRegion? = null

  fun load() {
    transparent = Singularity.getModDrawable<Drawable?>("transparent")
    sglLaunchLogo = Singularity.getModDrawable("launch_logo")
    squareMarker = Singularity.getModDrawable<Drawable?>("square_marker")
    matrixArrow = Singularity.getModDrawable<Drawable?>("matrix_arrow")
    sglIcon = Singularity.getModDrawable<Drawable?>("sgl_icon")
    artistIcon = Singularity.getModDrawable<Drawable?>("artist")
    codeIcon = Singularity.getModDrawable<Drawable?>("code")
    translateIcon = Singularity.getModDrawable<Drawable?>("translate")
    soundsIcon = Singularity.getModDrawable<Drawable?>("sound")
    time = Singularity.getModDrawable<Drawable?>("time")
    techPoint = Singularity.getModDrawable<Drawable?>("tech_point")
    inspire = Singularity.getModDrawable<Drawable?>("inspire")
    startIcon = Singularity.getModDrawable<Drawable?>("icon_start")
    databaseIcon = Singularity.getModDrawable<Drawable?>("icon_database")
    publicInfoIcon = Singularity.getModDrawable<Drawable?>("icon_publicInfo")
    aboutIcon = Singularity.getModDrawable<Drawable?>("icon_about")
    configureIcon = Singularity.getModDrawable<Drawable?>("icon_configure")
    contributeIcon = Singularity.getModDrawable<Drawable?>("icon_contribute")
    debuggingIcon = Singularity.getModDrawable<Drawable?>("debugging")
    nuclearIcon = Singularity.getModDrawable<Drawable?>("nuclear")
    matrixIcon = Singularity.getModDrawable<Drawable?>("matrix")
    qqIcon = Singularity.getModDrawable<Drawable?>("qq")
    telegramIcon = Singularity.getModDrawable<Drawable?>("telegram")
    showInfos = Singularity.getModDrawable<Drawable?>("show_infos")
    unShowInfos = Singularity.getModDrawable<Drawable?>("unshow_infos")
    showRange = Singularity.getModDrawable<Drawable?>("show_range")
    hold = Singularity.getModDrawable<Drawable?>("hold")
    defaultShow = Singularity.getModDrawable<Drawable?>("default_show")
    sgl2 = Singularity.getModDrawable<Drawable?>("sgl-2")
    a_z = Singularity.getModDrawable<Drawable?>("a_z")

    cursor = Singularity.getModAtlas("cursor")

    grayUI = (Tex.whiteui as TextureRegionDrawable).tint(Pal.darkerGray)
    padGrayUI = (Tex.whiteui as TextureRegionDrawable).tint(Pal.darkerGray)
    padGrayUI!!.setLeftWidth(8f)
    padGrayUI!!.setRightWidth(8f)
    padGrayUI!!.setTopHeight(8f)
    padGrayUI!!.setBottomHeight(8f)
    darkgrayUI = (Tex.whiteui as TextureRegionDrawable).tint(Pal.darkestGray)

    grayUIAlpha = (Tex.whiteui as TextureRegionDrawable).tint(Tmp.c1.set(Pal.darkerGray).a(0.7f))
    padGrayUIAlpha = (Tex.whiteui as TextureRegionDrawable).tint(Tmp.c1.set(Pal.darkerGray).a(0.7f))
    padGrayUIAlpha!!.setLeftWidth(8f)
    padGrayUIAlpha!!.setRightWidth(8f)
    padGrayUIAlpha!!.setTopHeight(8f)
    padGrayUIAlpha!!.setBottomHeight(8f)
    darkgrayUIAlpha = (Tex.whiteui as TextureRegionDrawable).tint(Tmp.c1.set(Pal.darkestGray).a(0.7f))
  }
}